web/src/filter.ts

```diff
@@ -1627,6 +1627,32 @@ export class Filter {
             });
     }
 
+    can_show_next_unread_topic_conversation_button(): boolean {
+        const term_type = this.sorted_term_types();
+        if (
+            _.isEqual(term_type, ["channel", "topic", "near"]) ||
+            _.isEqual(term_type, ["channel", "topic", "with"]) ||
+            _.isEqual(term_type, ["channel", "topic"]) ||
+            _.isEqual(term_type, ["channel"])
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    can_show_next_unread_dm_conversation_button(): boolean {
+        const term_type = this.sorted_term_types();
+        if (
+            _.isEqual(term_type, ["dm", "near"]) ||
+            _.isEqual(term_type, ["dm", "with"]) ||
+            _.isEqual(term_type, ["dm"]) ||
+            _.isEqual(term_type, ["is-dm"])
+        ) {
+            return true;
+        }
+        return false;
```

Comment:
This code with its tests would be a good thing to extract as a prep commit, to help shrink the overall PR and limit merge conflicts.

You should call your local variable `term_type_list` or something, though. Looks like we have that naming bug in the adjacent functions too, which may help make for a good little prep/cleanup PR.

Reply:
Opened #31680 using `term_types` as name since that was consistently used and seemed like an easier to read name.